warehouse/internal/repo/staging.go

```diff
@@ -317,10 +317,18 @@ func (repo *StagingFiles) GetAfterID(ctx context.Context, sourceID, destinationI
 }
 
 func (repo *StagingFiles) Pending(ctx context.Context, sourceID, destinationID string) ([]model.StagingFile, error) {
-	var lastStagingFileID int64
+	var (
+		uploadID               int64
+		lastStartStagingFileID int64
+		lastEndStagingFileID   int64
+		useUploadID            sql.NullBool
+	)
 	err := repo.db.QueryRowContext(ctx, `
 		SELECT
-			end_staging_file_id
+			id,
+			start_staging_file_id,
+			end_staging_file_id,
+			metadata->>'use_upload_id' AS use_upload_id
```

Comment:
Should we be fetching cross db inside repo meant for staging files ? Or should we move this call a layer above or have access to upload repo here ? Wdyt ?

Reply:
A good point. I believe `stagingFiles.Pending` is a good abstraction, the question I present myself is: 
Should the service layer know that in order to fetch pending events, we have to get the max staging file from another file? 

For me, this is an implementation detail specific to our persistent layer that needs to be abstracted.

Moreover, once the migration phase gets completed, we can fetch pending files by just looking at a single table.